birdhouse/read-configs.include.sh

```diff
@@ -108,6 +108,7 @@ source_conf_files() {
       if [ -f "$COMPONENT_DEFAULT_ENV" ]; then
           echo "reading '$COMPONENT_DEFAULT_ENV'"
           . "$COMPONENT_DEFAULT_ENV"
+          source_conf_files "$COMPONENT_DEPENDENCIES" "a dependency of $adir"
```

Comment:
I order to make this config loader order respect the intended override precedence, I would suggest the following logic change.

Current logic in `source_conf_files()`:
```
for adir in list_of_conf_dirs:
  add adir to ALL_CONF_DIRS
  read adir/default.env  # get value of COMPONENT_DEPENDENCIES
  recursive call source_conf_files on COMPONENT_DEPENDENCIES
    add COMPONENT_DEPENDENCIES to ALL_CONF_DIRS
    read defaut.env of COMPONENT_DEPENDENCIES  # get value of COMPONENT_DEPENDENCIES and turtle down
```

If `comp2` depends on `comp1`, and only `comp2` is enabled, `ALL_CONF_DIRS` will list `comp2` before `comp1` so if `comp2` has a fragment to override `comp1` compose file, the fragment might not work since `comp1` compose file has not been read yet.

Furthermore, if `comp2 default.env` has override for `comp1 default.env`, given `comp2 default.env` is read before `comp1 default.env`, it will not be able to override it properly.

Also, since the ordering of `ALL_CONF_DIRS` is reversed, we will also executing all the pre/post docker-compose-up scripts in the reverse order.

To be safe, maybe I would change the logic to something like below (untested, I might be wrong).

Proposed logic in source_conf_files():
```
for adir in list_of_conf_dirs:
  read adir/default.env  # get value of COMPONENT_DEPENDENCIES

  recursive call source_conf_files on COMPONENT_DEPENDENCIES
    read adir of COMPONENT_DEPENDENCIES default.env   # COMPONENT_DEPENDENCIES variable name clashing here, might need to save previous value

    recursive call source_conf_files on COMPONENT_DEPENDENCIES  # and turtle down 

    read adir COMPONENT_DEPENDENCIES default.env  # read again to be able to override dependent default.env
    append adir COMPONENT_DEPENDENCIES to ALL_CONF_DIRS

  read adir/default.env  # again to override dependent default.env
  append adir to ALL_CONF_DIRS
```

So the suggested logic above might fix the loading order for "hard" dependencies (listed in `COMPONENT_DEPENDENCIES`).

For "soft" dependencies (ex: `all-public-access` has magpie config to make all the birds public but do not list all the birds in its `COMPONENT_DEPENDENCIES`), the user still need to ensure `all-public-access` is listed after all the birds (all its soft dependencies) in `EXTRA_CONF_DIRS`.